app/controllers/orders_controller.rb

```diff
@@ -9,16 +9,29 @@ def create
         cart_items = current_user.cart.cart_items
         
         @order = Order.new(user: current_user)
+        products_to_update = []
+
         total = calculate_total_order_value(cart_items)
         cart_items.each do |cart_item|
             product = Product.find(cart_item[:product_id])
+             if cart_item.quantity > product.stock_quantity
+                # If any item's quantity exceeds the stock quantity, cancel the order and throw an error
+                @order.destroy
+                flash[:error] = "The quantity of #{product.name} exceeds the available stock."
+                redirect_to root_path
+                return
+            end
+            product.stock_quantity -= cart_item.quantity
             @order.order_items.build(product: product, quantity: cart_item[:quantity], total: cart_item.quantity * product.price)
+            products_to_update << product
```

Comment:
What is the use of products_to_update in this action.

Reply:
i am storing this product_to_update stores the product details with updated stock quantity ,by refering the order_items quantity,but products has not been saved yet,
The product only gets save once the order gets confirmed ,else we don;t update product stock quantity